homeassistant/components/nexia/climate.py

```diff
@@ -224,20 +234,48 @@ def preset_mode(self):
         return self._zone.get_preset()
 
     async def async_set_humidity(self, humidity: int) -> None:
-        """Dehumidify target."""
-        if self._thermostat.has_dehumidify_support():
-            await self.async_set_dehumidify_setpoint(humidity)
+        """Set humidity targets.
+
+        HA doesn't support separate humidify and dehumidify targets.
+        Set the target for the current mode if in [heat, cool]
+        otherwise set both targets to the clamped values.
+        """
+
+        if self._zone.get_current_mode() == OPERATION_MODE_HEAT:
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+        elif self._zone.get_current_mode() == OPERATION_MODE_COOL:
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_dehumidify_setpoint(humidity)
         else:
-            await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_humidify_support():
+                await self.async_set_humidify_setpoint(humidity)
+            if self._thermostat.has_dehumidify_support():
+                await self.async_set_dehumidify_setpoint(humidity)
         self._signal_thermostat_update()
 
     @property
-    def target_humidity(self):
-        """Humidity indoors setpoint."""
+    def target_humidity(self) -> float | None:
+        """Humidity indoors setpoint.
+
+        In systems that support both humidification and dehumidification,
+        two values for target exist. We must choose one to return.
+
+        :return: The target humidity setpoint.
+        """
+
+        # If heat is on, always return humidify value first
+        if (
+            self._zone.get_current_mode() == OPERATION_MODE_HEAT
+            and self._has_humidify_support
```

Comment:
Nit: Please reverse the check order here.  Better to check the var before doing the comparison operation